daphne/src/testing.rs

```diff
@@ -470,32 +474,156 @@ impl DapLeader<BearerToken> for MockAggregator {
         Ok(HashMap::from([(selector.task_id.clone(), reports)]))
     }
 
-    async fn init_collect_job(&self, _collect_req: &CollectReq) -> Result<Url, DapError> {
-        unreachable!("not implemented");
+    // Called after receiving a CollectReq from Collector.
+    async fn init_collect_job(&self, collect_req: &CollectReq) -> Result<Url, DapError> {
+        let mut rng = thread_rng();
+        let task_config = self
+            .get_task_config_for(&collect_req.task_id)
+            .ok_or_else(|| DapError::fatal("no task found"))?;
+
+        let mut leader_state_store_mutex_guard = self
+            .leader_state_store
+            .lock()
+            .map_err(|e| DapError::Fatal(e.to_string()))?;
+        let leader_state_store = leader_state_store_mutex_guard.deref_mut();
+
+        // Store Collect ID and CollectReq into LeaderState.
+        let collect_id = Id(rng.gen());
+        if !leader_state_store.contains_key(&collect_req.task_id) {
+            leader_state_store.insert(collect_req.task_id.clone(), LeaderState::new());
+        }
+
+        let leader_state = leader_state_store.get_mut(&collect_req.task_id).unwrap();
+        leader_state.collect_ids.push_back(collect_id.clone());
+        let collect_job_state = CollectJobState {
+            collect_req: Some(collect_req.clone()),
+            collect_resp: None,
+        };
+        leader_state
+            .collect_jobs
+            .insert(collect_id.clone(), collect_job_state);
```

Comment:
This clone is also avoidable.

Reply:
Ditto.

Reply:
This clone can be avoided by moving `collect_id.to_base64url()`, above this point and moving `collect_id` into `.insert()` here.